sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -597,12 +599,26 @@ class ParquetFilters(
         createFilterHelper(pred, canPartialPushDownConjuncts = false)
           .map(FilterApi.not)
 
-      case sources.In(name, values) if canMakeFilterOn(name, values.head)
-        && values.distinct.length <= pushDownInFilterThreshold =>
-        values.distinct.flatMap { v =>
-          makeEq.lift(nameToParquetField(name).fieldType)
-            .map(_(nameToParquetField(name).fieldNames, v))
-        }.reduceLeftOption(FilterApi.or)
+      case sources.In(name, values) if pushDownInFilterThreshold > 0 &&
```

Comment:
@wangyum, the impala reference sounds good. Can we make it general and push the range filter to other data sources as well?

Reply:
If this is supposed to be beneficial in other sources as well, I think it makes more sense to push it to other sources as well anyway.

Reply:
It seems only Parquet is not well supported `In` predicate pushdown.
Parquet vs ORC: https://github.com/apache/spark/blob/f5118f81e395bde0cd8253dbef6a9e6455c3958a/sql/core/benchmarks/FilterPushdownBenchmark-results.txt#L439-L482
CSV:
https://github.com/apache/spark/pull/29642#issuecomment-730869008